### PR TITLE
Kaltura support

### PR DESCRIPTION
**Pull request by ilyaDymshits**

### CODE REVIEW

File: amd/src/annoto.js [hunk 1]
```diff
@@ -183,6 +196,112 @@ define([
...
+                features: {
+                    tabs: params.featureTab,
+                    cta: params.featureCTA,
+                },
+                width: {
+                    max: (horizontalAlign === 'inner') ? 320 : 360,
+                },
+                align: {
+                    vertical: params.alignVertical,
+                    horizontal: horizontalAlign,
```
Comment (genadis) [inline amd/src/annoto.js]: @ilyaDymshits remove the `horizontalAlign` config (see the notes above). keep only the `align.vertical`

File: amd/src/annoto.js [hunk 2]
```diff
@@ -183,6 +196,112 @@ define([
...
+                },
+                ux: {
+                    ssoAuthRequestHandle: function() {
+                        window.location.replace(params.loginUrl);
+                    },
+                },
+                zIndex: params.zIndex ? params.zIndex : 100,
```
Comment (genadis) [inline amd/src/annoto.js]: @ilyaDymshits `zIndex` is not required for the kaltura player. please remove.

File: amd/src/annoto.js [hunk 3]
```diff
@@ -183,6 +196,112 @@ define([
...
+                demoMode: params.demoMode,
+                rtl: params.rtl,
+                locale: params.locale,
+            };
+
+            Object.assign(config, configSrc);
```
Comment (genadis) [inline amd/src/annoto.js]: @ilyaDymshits Couple of issues with this technique:

1. Object.assign is not supported by IE11
2. Object.assign does not do deep merge. so all the objects would be overwritten.

Instead use this (solves the other issues related to config mentioned above):
```javascript
var widget = config.widgets[0];
var ux = config.ux || {};
var align = config.align || {};
var features = config.features || {};

config.ux = ux;
config.align = align;
config.features = features;

config.clientId = params.clientId;
config.position = params.position;
config.demoMode = params.demoMode;
config.locale = params.locale;
config.rtl = params.rtl;

features.tabs = params.featureTab;
features.cta = params.featureCTA;
align.vertical = params.alignVertical;
ux.ssoAuthRequestHandle = function() {
    window.location.replace(params.loginUrl);
}
playerConfig.mediaDetails = this.enrichMediaDetails.bind(this);
```

File: amd/src/annoto.js [hunk 4]
```diff
@@ -183,6 +196,112 @@ define([
...
+             * https://github.com/Annoto/moodle-local_annoto/blob/master/amd/src/annoto.js#L167
+             * below is simplified version.
+             */
+            pluginApi.auth(this.params.userToken).catch(function() {
+                log.error('Annoto SSO auth error');
+            });
+
+        },
+
+        annotoSetup: function(config) {
```
Comment (genadis) [inline amd/src/annoto.js]: @ilyaDymshits this function is dedicated for the kaltura config. change the name to `setupKalturaPlugin`

File: amd/src/annoto.js [hunk 5]
```diff
@@ -183,6 +196,112 @@ define([
...
+             * Annoto Kaltura plugin, already has some details about the media like title.
+             * But for moodle if the title and description we get from Moodle is the activity and present,
+             * we should override it, if it's embedded in places where there is no Moodle media title, don't change it.
+             * The group should always be set as done here:
+             * https://github.com/Annoto/moodle-local_annoto/blob/master/amd/src/annoto.js#L142
+             *
+             * retVal.group = { ... };
+             */
+
+            var retVal = Object.assign(details, mediaDetails);
```
Comment (genadis) [inline amd/src/annoto.js]: @ilyaDymshits Does Moodle include polyfill for `Oeject.assign` for IE11? If not, don't use it.

**Comment by ilyaDymshits** [comment]
הי  !
קיבלתי, תודה  , אעבור על זה

איליה

File: lib.php [hunk 6]
```diff
@@ -24,10 +24,15 @@
...
 
 /**
- * Callback function - injects Annoto's JS into every page.
+ * Allow plugins to provide some content to be rendered in the navbar.
+ * The plugin must define a PLUGIN_render_navbar_output function that returns
+ * the HTML they wish to add to the navbar.
+ *
+ * @return string HTML for the navbar
  */
-function local_annoto_before_footer() {
```
Comment (genadis) [inline lib.php]: @ilyaDymshits Shouldn't this be reverted back to `local_annoto_before_footer` or something? or is there timing issue related to kaltura?

Comment (ilyaDymshits) [reply lib.php]: needs to check it. according to the moodle documentation if we wont to call js on <head> we should use requare->js('script', true) with second parameter true, but for some reason, the page renderer is falling down.

**Pull request merged**